Source/towners.cpp

```diff
@@ -103,1187 +98,1024 @@ char AnimOrder[6][148] = {
-	for (i = 0; i < 8; i++) {
-		towners[numtowners]._tNAnim[i] = towners[numtowners]._tNData.get();
-	}
-	towners[numtowners]._tNFrames = 25;
-	NewTownerAnim(numtowners, towners[numtowners]._tNAnim[DIR_S], towners[numtowners]._tNFrames, 3);
-	towners[numtowners]._tName = _("Cain the Elder");
-	numtowners++;
+	InitQstSnds(towner, initData.type);
+	towner._tAnimWidth = 96;
+	towner._tAnimOrder = -1;
+	LoadTownerAnimations(towner, "Towners\\Farmer\\Farmrn2.CEL", 15, initData.dir, 3);
+	towner._tName = _("Lester the farmer");
 }
 
-void InitDrunk()
+void InitCowFarmer(TownerStruct &towner, const TownerInit &initData)
 {
-	int i;
-
-	InitTownerInfo(numtowners, 96, true, TOWN_DRUNK, 71, 84, 4);
-	InitQstSnds(numtowners, TOWN_DRUNK);
-	towners[numtowners]._tNData = LoadFileInMem("Towners\\Drunk\\TwnDrunk.CEL");
-	for (i = 0; i < 8; i++) {
-		towners[numtowners]._tNAnim[i] = towners[numtowners]._tNData.get();
+	InitQstSnds(towner, initData.type);
+	const char *celPath = "Towners\\Farmer\\cfrmrn2.CEL";
+	if (quests[Q_JERSEY]._qactive == QUEST_DONE) {
+		celPath = "Towners\\Farmer\\mfrmrn2.CEL";
 	}
-	towners[numtowners]._tNFrames = 18;
-	NewTownerAnim(numtowners, towners[numtowners]._tNAnim[DIR_S], towners[numtowners]._tNFrames, 3);
-	towners[numtowners]._tName = _("Farnham the Drunk");
-	numtowners++;
+	towner._tAnimWidth = 96;
+	towner._tAnimOrder = -1;
+	LoadTownerAnimations(towner, celPath, 15, initData.dir, 3);
+	towner._tName = _("Complete Nut");
 }
 
-void InitCows()
+void InitGirl(TownerStruct &towner, const TownerInit &initData)
 {
-	int i, dir;
-	int x, y, xo, yo;
-
-	//if ( pCowCels )
-	// assertion_failed(__LINE__, __FILE__, "! pCowCels");
-	pCowCels = LoadFileInMem("Towners\\Animals\\Cow.CEL");
-	for (i = 0; i < 3; i++) {
-		x = TownCowX[i];
-		y = TownCowY[i];
-		dir = TownCowDir[i];
-		InitTownerInfo(numtowners, 128, false, TOWN_COW, x, y, -1);
-		towners[numtowners]._tNData = nullptr;
-		SetTownerGPtrs(pCowCels.get(), towners[numtowners]._tNAnim);
-		towners[numtowners]._tNFrames = 12;
-		NewTownerAnim(numtowners, towners[numtowners]._tNAnim[dir], towners[numtowners]._tNFrames, 3);
-		towners[numtowners]._tAnimFrame = GenerateRnd(11) + 1;
-		towners[numtowners]._tSelFlag = true;
-		towners[numtowners]._tName = _("Cow");
-
-		xo = x + cowoffx[dir];
-		yo = y + cowoffy[dir];
-		if (dMonster[x][yo] == 0)
-			dMonster[x][yo] = -(numtowners + 1);
-		if (dMonster[xo][y] == 0)
-			dMonster[xo][y] = -(numtowners + 1);
-		if (dMonster[xo][yo] == 0)
-			dMonster[xo][yo] = -(numtowners + 1);
-
-		numtowners++;
+	InitQstSnds(towner, initData.type);
+	const char *celPath = "Towners\\Girl\\Girlw1.CEL";
+	if (quests[Q_GIRL]._qactive == QUEST_DONE) {
+		celPath = "Towners\\Girl\\Girls1.CEL";
 	}
+	towner._tAnimWidth = 96;
+	towner._tAnimOrder = -1;
+	LoadTownerAnimations(towner, celPath, 20, initData.dir, 6);
+	towner._tName = "Celia";
 }
 
-void InitFarmer()
+TownerInit TownerInitList[] = {

```

Comment:
Can this be `const`?

Reply:
Defiantly, don't want to make the same mistake that the original does twice.